src/jwplayer/players/shared/JwPlayerWrapperWithStrategyRules.tsx

```diff
@@ -0,0 +1,169 @@
+			const dismissed = getDismissed();
+			// Pause the content play when the user closed the mini player playing the ad
+			if (dismissed && viewable === 0 && playReason === 'autostart') {
+				playerInstance.pause();
+			}
+		});
+
+		playerInstance.on(JWEvents.READY, () => {
+			recordVideoEvent(VIDEO_RECORD_EVENTS.JW_PLAYER_READY);
+			triggerVideoMetric('ready');
+			// only add the events after the player is ready
+			jwPlayerPlaybackTracker({ event_name: 'video_player_ready' });
+			addBaseTrackingEvents(playerInstance);
+
+			if (onReady) {
+				console.debug('Player ready...');
+				onReady(playerInstance);
+			}
+		});
+
+		playerInstance.on(JWEvents.COMPLETE, () => {
+			if (typeof onComplete === 'function') {
+				onComplete();
+			}
+
+			// Incrementing videos watched count
+			videoIndexRef.current += 1;
+
+			// if stopAutoAdvanceOnExitViewport is false then we don't want to stop the auto advance, keep normal behavior
+			if (!stopAutoAdvanceOnExitViewport) {
+				return;
+			}
+		});
+	};
+
+	const isJWPlayerReady = () => {
+		return typeof window.jwplayer === 'function';
+	};
+
+	const jwPlayerLoaded = () => {
+		console.debug('Player loaded!', jwPlayerContainerEmbedId);
+
+		// Set the max_resolution param for related videos
+		if (typeof window?.jwplayer?.defaults?.related?.file === 'string') {
+			window.jwplayer.defaults.related.file = window.jwplayer.defaults.related.file + '&max_resolution=1280';
+		}
+
+		jwPlayerPlaybackTracker({ event_name: 'video_player_load' });
+		recordVideoEvent(VIDEO_RECORD_EVENTS.JW_PLAYER_SCRIPTS_LOAD_READY);
+		triggerVideoMetric('loaded');
+
+		setConfig(config);
+
+		// TODO: refactor this so we get here the container within jwPlayerContainerEmbedId
+		const playerInstance = window.jwplayer();
+		console.debug('playerInstance: ', playerInstance);
+		registerEventHandlers(playerInstance);
+		setPlayer(playerInstance);
+	};
+
+	const initPlayer = () => {
+		console.debug('initPlayer...');
+		recordVideoEvent(VIDEO_RECORD_EVENTS.JW_PLAYER_SCRIPTS_LOAD_START);
+		jwPlayerPlaybackTracker({ event_name: 'video_player_start_load' });
+
+		const onload = () => {
+			console.debug('Strategy rules embed loaded. Waiting for player...');
+
+			const waitForJWPlayer = waitFor({ eventCheck: isJWPlayerReady });
+			waitForJWPlayer().then(jwPlayerLoaded, () => {
+				console.error('JWPlayer not ready', window?.jwplayer);
+			});
+		};
+
+		if (isJWPlayerReady()) {
+			onload();
+		} else {
+			const script = document.createElement('script');
+			script.async = true;
+			script.src = 'https://cdn.jwplayer.com/v2/sites/cGlKNUnj/placements/embed.js';
+			script.onload = onload;
+			document.getElementsByTagName('head')[0].appendChild(script);
+		}
+	};
+
+	return (
+		<div className={className}>
+			<div
+				id={jwPlayerContainerEmbedId}
+				data-jw-placement-id={'21rL5wJF'}
+				data-jw-playlist={config.playlistUrl}
+				data-jw-recommendations_playlist_id={'FOhaD53w'}
```

Comment:
I'm curious if it would be a good idea to place the placement ID and playlist ID in a configuration file or, at the very least, store them as constant variables.

Reply:
ACK. These will be dynamic in the future most likely 😊